src/neuroconv/utils/json_schema.py

```diff
@@ -36,45 +31,38 @@ def default(self, obj):
         if isinstance(obj, np.generic):
             return obj.item()
 
+        # Numpy arrays should be converted to lists
         if isinstance(obj, np.ndarray):
             return obj.tolist()
 
+        # Over-write behaviors for Paths
+        if isinstance(obj, Path):
+            return str(obj)
+
         # The base-class handles it
         return super().default(obj)
 
 
-class _NWBSourceDataEncoder(_NWBMetaDataEncoder):
+class _NWBMetaDataEncoder(_GenericNeuroconvEncoder):
     """
-    Custom JSON encoder for data interface source data (i.e. kwargs).
-
-    This encoder extends the default JSONEncoder class and provides custom serialization
-    for certain data types commonly used in interface source data.
+    Custom JSON encoder for NWB metadata.
     """
 
-    def default(self, obj):
 
-        # Over-write behaviors for Paths
-        if isinstance(obj, Path):
-            return str(obj)
-
-        return super().default(obj)
+class _NWBSourceDataEncoder(_GenericNeuroconvEncoder):
+    """
+    Custom JSON encoder for data interface source data (i.e. kwargs).
+    """
 
 
-class _NWBConversionOptionsEncoder(_NWBMetaDataEncoder):
+class _NWBConversionOptionsEncoder(_GenericNeuroconvEncoder):
     """
```

Comment:
I think this is a strange pattern. Why define child classes that are identical to the parent class? Why not just use the parent class?

Reply:
They might diverge and we might not want the same behavior for each of those cases.

Reply:
Given that they are private though, I don't mind having the simple case as you want it and then refactoring if and when we need it.